File: SDK/php/src/Exceptions/RateLimitException.php
<?php

namespace Verifio\Exceptions;

class RateLimitException extends VerifioException
{
    protected ?int $retryAfter;

    public function __construct(?string $message = null, ?int $retryAfter = null)
    {
        parent::__construct($message ?? "Rate limit exceeded", 429, "RATE_LIMIT_EXCEEDED");
        $this->retryAfter = $retryAfter;
    }

    public function getRetryAfter(): ?int
    {
        return $this->retryAfter;
    }
}

File: SDK/java/src/main/java/com/reloop/verifio/errors/RateLimitException.java
package com.reloop.verifio.errors;

public class RateLimitException extends VerifioException {
    private final Integer retryAfter;

    public RateLimitException(String message, Integer retryAfter) {
        super(message != null ? message : "Rate limit exceeded", 429, "RATE_LIMIT_EXCEEDED");
        this.retryAfter = retryAfter;
    }

    public Integer getRetryAfter() {
        return retryAfter;
    }
}
